src/cloud-api-adaptor/pkg/securecomms/kubemgr/kubemgr.go

```diff
@@ -0,0 +1,127 @@
+package kubemgr
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"flag"
+	"fmt"
+	"log"
+	"path/filepath"
+
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshutil"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var KubeMgr *KubeMgrStruct
+
+const (
+	cocoNamespace = "confidential-containers-system"
+)
+
+type KubeMgrStruct struct {
+	client        *kubernetes.Clientset
+	cocoNamespace string
+}
+
+var SkipVerify bool
+
+func InitKubeMgr() error {
+	var err error
+	KubeMgr = &KubeMgrStruct{
+		cocoNamespace: cocoNamespace,
+	}
+
+	var kubeCfg *rest.Config
+	var devKubeConfigStr *string
+
+	// Try to detect in-cluster config
+	if kubeCfg, err = rest.InClusterConfig(); err != nil {
+		// Not running in cluster
+		if home := homedir.HomeDir(); home != "" {
+			devKubeConfigStr = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+		} else {
+			devKubeConfigStr = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+		}
+		flag.Parse()
+
+		// Use the current context in kubeconfig
+		if kubeCfg, err = clientcmd.BuildConfigFromFlags("", *devKubeConfigStr); err != nil {
+			return fmt.Errorf("no Config found to access KubeApi! err: %w", err)
+		}
+	}
+
+	// Create a secrets client
+	if SkipVerify {
+		kubeCfg.TLSClientConfig.Insecure = true
+		kubeCfg.TLSClientConfig.CAData = nil
+	}
+
+	KubeMgr.client, err = kubernetes.NewForConfig(kubeCfg)
+	if err != nil {
+		return fmt.Errorf("failed to configure KubeApi using config: %w", err)
+	}
+	return nil
+}
+
+func (kubeMgr *KubeMgrStruct) ReadSecret(secretName string) (privateKey []byte, publicKey []byte, err error) {
+	secrets := kubeMgr.client.CoreV1().Secrets(kubeMgr.cocoNamespace)
+	secret, err := secrets.Get(context.Background(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	privateKey = secret.Data["privateKey"]
+	publicKey = secret.Data["publicKey"]
+	return
+}
+
+func (kubeMgr *KubeMgrStruct) DeleteSecret(secretName string) {
+	secrets := kubeMgr.client.CoreV1().Secrets(kubeMgr.cocoNamespace)
+	secrets.Delete(context.Background(), secretName, metav1.DeleteOptions{})
+}
+
+func (kubeMgr *KubeMgrStruct) CreateSecret(secretName string) (privateKey []byte, publicKey []byte, err error) {
```

Comment:
Is there any plan to give users ability to inject an existing key? Any use case for that? Might be worth adding your answer to the documentation / cover letter.

Reply:
No plan to enable user injected keys. Yet, if keys already exist, the implementation read such keys and use them before creating new ones. As of today, it is expected that keys will always be created by the system and not externally by a user.